druid/examples/hello_wasm_web/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use druid::widget::{Align, Flex, Label, TextBox};
+use druid::{AppLauncher, Data, Env, Lens, LocalizedString, Widget, WidgetExt, WindowDesc};
+
+use wasm_bindgen::prelude::*;
+
+const VERTICAL_WIDGET_SPACING: f64 = 20.0;
+const TEXT_BOX_WIDTH: f64 = 200.0;
+const WINDOW_TITLE: LocalizedString<HelloState> = LocalizedString::new("Hello World!");
+
+#[derive(Clone, Data, Lens)]
+struct HelloState {
+    name: String,
+}
+
+// This wrapper function is the only modification we're making to the vanilla
+// hello.rs example.
+#[wasm_bindgen]
+pub fn wasm_main() {
+    std::panic::set_hook(Box::new(console_error_panic_hook::hook));
```

Comment:
Would be nice if this one could get a comment, explaining that this is necessary to get panic messages on wasm32.

Reply:
I was unaware of what this actually did, so I hope my copy of your explanation is sufficient.